### PR TITLE
Untrunctaed normal distribution in VarianceScalingInitializer

### LINKED ISSUE #19996: Feature Request: untruncated normal in variance scaling initializer
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: N/A
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: N/A
- **TensorFlow installed from (source or binary)**: N/A
- **TensorFlow version (use command below)**: N/A
- **Python version**: N/A
- **Bazel version (if compiling from source)**:N/A
- **GCC/Compiler version (if compiling from source)**:N/A
- **CUDA/cuDNN version**:N/A
- **GPU model and memory**:N/A
- **Exact command to reproduce**:N/A

Currently the "normal" mode of `variance_scaling_initializer` actually produces truncated normal distribution. 
Despite that I saw Google suggest in several places that a truncated normal might be better than a true normal distribution, truncated normal is NOT what is used in the initialization literature such as:

[1] Glorot, Xavier, and Yoshua Bengio. "Understanding the difficulty of training deep feedforward neural networks." In Proceedings of the thirteenth international conference on artificial intelligence and statistics, pp. 249-256. 2010.
[2] He, Kaiming, Xiangyu Zhang, Shaoqing Ren, and Jian Sun. "Delving deep into rectifiers: Surpassing human-level performance on imagenet classification." In Proceedings of the IEEE international conference on computer vision, pp. 1026-1034. 2015.

I found that this difference is affecting some of my experiments. To be more consistent with literature, I would request a new mode in variance_scaling_initializer that produces untruncated normal distribution. I'm OK to implement it If someone can give it a good name.

### PR DESCRIPTION
**Pull request by ppwwyyxx**
Fixes #19996 

RELNOTES: VarianceScalingInitializer now takes "truncated_normal" or "untruncated_normal" as values for its distribution argument. Using "normal" is equivalent to "truncated_normal" (the current behavior), but its use is deprecated. 

@martinwicke

### CODE REVIEW

File: tensorflow/python/ops/init_ops.py [hunk 1]
```diff
@@ -419,6 +421,8 @@ class VarianceScaling(Initializer):
   With `distribution="uniform"`, samples are drawn from a uniform distribution
   within [-limit, limit], with `limit = sqrt(3 * scale / n)`.
 
+  `distribution="normal"` is a deprecated alias for "truncated_normal".
```
Comment (martinwicke) [inline tensorflow/python/ops/init_ops.py]: If you want, you can use @deprecated_arg_values to formally encode this (with a warning). But I think what you did here is fine.

**Review by martinwicke — APPROVED** [review]

**Review by martinwicke — CHANGES_REQUESTED** [review]
Actually, we should have a test. I'd mock out random_normal and truncated_normal and check that the right one is called (I think a statistical tests would be expensive or prone to be flaky, but if you can make one that isn't, go for it).

**Review by martinwicke — CHANGES_REQUESTED** [review]
If you find that these random ops are already seeded properly, there's nothing to do (may be done in test_session).

**Review by martinwicke — DISMISSED** [review]

**Comment by ppwwyyxx** [comment]
It turns out that `assert_called()` is only available in Py3.6+

**Comment by ppwwyyxx** [comment]
Now the failures are: 
`//tensorflow/tools/api/tests:api_compatibility_test`. I'm not sure how that works or if I should do anything to make it pass.
`//tensorflow/core:common_runtime_ring_reducer_test`: that's probably unrelated to this PR.

**Comment by martinwicke** [comment]
The api_compatibility test log has instructions on how to update the api goldens that it compares against. Please run that, you'll have to do that on MacOS or Linux, and I believe in Py2. 

@gunan is this still true? Do we have instructions for this somewhere (maybe we should).

**Review by martinwicke — APPROVED** [review]

**Comment by martinwicke** [comment]
Thank for the quick PR!

**Comment by gunan** [comment]
Sorry, I have not gotten around to making the test work on python 3, so it still  only works on python 2, on linux and macos. The test prints its instructions, and all instructions should already exist in the folder.

**Comment by ppwwyyxx** [comment]
Apart from rebuilding with py2, another painful point is that the linter script will always lint all thousands of python sources, despite the "--incremental" option (which is explicitly not used by the linter).

**Pull request merged**